misc/bazel/pkg.bzl

```diff
@@ -162,30 +209,39 @@ _imported_zips_manifest = rule(
         {CODEQL_PLATFORM} can be used as zip prefixes and will be expanded to the relevant codeql platform.
     """,
     attrs = {
-        "zips": attr.label_keyed_string_dict(
-            doc = "mapping from zip files to install prefixes",
-            allow_files = [".zip"],
+        "srcs": attr.label_list(
+            doc = "mappings from zip files to install prefixes in _ZipInfo format",
+            providers = [_ZipInfo],
         ),
-    } | _PLAT_DETECTION_ATTRS,
+    },
 )
 
 def _zipmerge_impl(ctx):
     zips = []
-    filename = ctx.attr.zip_name + "-"
-    platform = _detect_platform(ctx)
-    filename = "%s-%s.zip" % (ctx.attr.zip_name, platform if ctx.attr.kind == "arch" else "generic")
-    output = ctx.actions.declare_file(filename)
-    args = [output.path, "--prefix=%s" % ctx.attr.prefix, ctx.file.base.path]
-    for zip, prefix in ctx.attr.zips.items():
-        zip_kind, expanded_prefix = _expand_path(prefix, platform)
-        if zip_kind == ctx.attr.kind:
-            args.append("--prefix=%s/%s" % (ctx.attr.prefix, expanded_prefix.rstrip("/")))
-            args += [f.path for f in zip.files.to_list()]
-            zips.append(zip.files)
+    transitive_zips = []
+    output = ctx.actions.declare_file(ctx.attr.out)
+    args = [output.path]
+    for zip_target in ctx.attr.srcs:
+        if _ZipInfo in zip_target:
+            zip_info = zip_target[_ZipInfo]
+            for zip, prefix in zip_info.zips_to_prefixes.items():
+                args += [
+                    "--prefix=%s/%s" % (ctx.attr.prefix, prefix.rstrip("/")),
+                    zip.path,
+                ]
+                zips.append(zip)
+        else:
+            zips = zip_target.files.to_list()
```

Comment:
I don't see how this works if the srcs are mixed, some with a `ZipInfo` provider, some not, as this overwrites `zips` hard, and the other code path for `ZipInfo` providers doesn't set `transitive_zips` at all

Reply:
ouch, the zips rewriting is indeed a bug, thanks for catching this!

Reply:
I also don't fully understand how this was working at all though :confused: